shanoir-ng-datasets/src/main/java/org/shanoir/ng/studycard/model/condition/StudyCardDICOMConditionOnDatasets.java

```diff
@@ -0,0 +1,300 @@
+/**
+ * Shanoir NG - Import, manage and share neuroimaging data
+ * Copyright (C) 2009-2019 Inria - https://www.inria.fr/
+ * Contact us on https://project.inria.fr/shanoir/
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+package org.shanoir.ng.studycard.model.condition;
+
+import java.math.BigDecimal;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dcm4che3.data.Attributes;
+import org.dcm4che3.data.Keyword;
+import org.dcm4che3.data.StandardElementDictionary;
+import org.dcm4che3.data.VR;
+import org.shanoir.ng.dataset.model.Dataset;
+import org.shanoir.ng.datasetacquisition.model.DatasetAcquisition;
+import org.shanoir.ng.download.AcquisitionAttributes;
+import org.shanoir.ng.download.ExaminationAttributes;
+import org.shanoir.ng.download.WADODownloaderService;
+import org.shanoir.ng.shared.exception.PacsException;
+import org.shanoir.ng.studycard.model.DicomTagType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+import jakarta.persistence.DiscriminatorValue;
+import jakarta.persistence.Entity;
+
+@Entity
+@Component
+@DiscriminatorValue("StudyCardDICOMConditionOnDatasets")
+@JsonTypeName("StudyCardDICOMConditionOnDatasets")
+public class StudyCardDICOMConditionOnDatasets extends StudyCardCondition implements ApplicationContextAware {
+	
+    private static final Logger LOG = LoggerFactory.getLogger(StudyCardDICOMConditionOnDatasets.class);
+    
+	private int dicomTag;
+
+    private static ApplicationContext context;
```

Comment:
Hi @julien-louis, I do not think the AppContext is necessary here, to discuss, what the best solution might be, but this might be to change: WADODownloaderService downloader = (WADODownloaderService) context.getBean("WADODownloaderService");. The reference to the static context is complicated and the bean look up, takes time as well (even if not much probably). We should find a way please to use @Autowired

Reply:
and move some logic to a separate service, that deals with the entity